src/flags/orgFlags.ts

```diff
@@ -0,0 +1,110 @@
+/*
+ * Copyright (c) 2020, salesforce.com, inc.
+ * All rights reserved.
+ * Licensed under the BSD 3-Clause license.
+ * For full license text, see LICENSE.txt file in the repo root or https://opensource.org/licenses/BSD-3-Clause
+ */
+import { Flags } from '@oclif/core';
+import { Messages, Org, ConfigAggregator } from '@salesforce/core';
+
+Messages.importMessagesDirectory(__dirname);
+const messages = Messages.loadMessages('@salesforce/sf-plugins-core', 'messages');
+
+const maybeGetOrg = async (input?: string): Promise<Org | undefined> => Org.create({ aliasOrUsername: input });
+const getOrgOrThrow = async (input?: string): Promise<Org> => {
+  const org = await maybeGetOrg(input);
+  if (!org) {
+    throw messages.createError('errors.NoDefaultEnv');
+  }
+  return org;
+};
+
+const getHubOrThrow = async (aliasOrUsername?: string): Promise<Org> => {
+  if (!aliasOrUsername) {
+    // check config for a default
+    const config = await ConfigAggregator.create();
+    aliasOrUsername = config.getInfo('defaultdevhubusername')?.value as string;
+    if (!aliasOrUsername) {
+      throw messages.createError('errors.NoDefaultHub', [aliasOrUsername]);
+    }
+  }
+  const org = await Org.create({ aliasOrUsername });
+  // check the synchronous, cached version and only use "determine" if we didn't get a positive result
+  if (!org.isDevHubOrg() || !(await org.determineIfDevHubOrg())) {
+    throw messages.createError('errors.NotADevHub', [aliasOrUsername]);
+  }
+  return org;
+};
+
+/**
+ * An optional org specified by username or alias
+ * Will default to the default org if one is not specified.
+ * Will not throw if the specified org and default do not exist
+ *
+ * @example
+ * import { SfCommand, optionalOrgFlag } from '@salesforce/sf-plugins-core';
+ * public static flags = {
+ *     // setting length or prefix
+ *    'target-org': optionalOrgFlag(),
+ *    // adding properties
+ *    'flag2': optionalOrgFlag()({
```

Comment:
This syntax is strange to me. Can we make it so that they can pass in their options like this:

```
flag2: optionOrgFlag({
  required: true,
  description: 'flag2 description',
})
```

Reply:
oh, to make the custom properties extend the oclif ones?  And then try to merge them?

Reply:
Yeah, that's what I was thinking